app/models/appointment_form.rb

```diff
@@ -8,4 +8,18 @@ class AppointmentForm
   def initialize(hash = {})
     @appointment_id = hash[:appointment_id]
   end
+
+  def begin_date
+    split_date.first
+  end
+
+  def end_date
+    split_date.last
+  end
+
+  private
+
+  def split_date
+    @split_date ||= @appointment_id.to_s.split('|')
```

Comment:
Is this because the ID will be of form `firstdate|lastdate`? Does this need to be shown in a test somewhere?

Reply:
Yeah, this exists because appointments don't have IDs - in fact, we still have no idea whether start/end time is enough to book with - I will add some tests around this.